datadog_checks_base/datadog_checks/base/stubs/aggregator.py

```diff
@@ -244,6 +245,26 @@ def assert_metric_has_tag_prefix(self, metric_name, tag_prefix, count=None, at_l
         else:
             assert len(candidates) >= at_least
 
+    def _match_metric_name_in_mapping(self, name, at_least):
```

Comment:
Can we solve this instead by changing our fixtures?

Reply:
@ofek our current fixtures are legit prometheus payloads, I don't think we should change them otherwise we could break submitted metric names.

This change in the library is actually breaking a few projects because it enforces `counter` to end up with `_total` see https://github.com/matrix-org/synapse/issues/4001 or https://github.com/korfuri/django-prometheus/issues/84

breaking change: https://github.com/prometheus/client_python/pull/300/commits/a4dd93bcc6a0422e10cfa585048d1813909c6786#diff-0adf47ea7f99c66d4866ccb4e557a865L158

Reply:
Thanks @ofek for the review, I admit this solution is a bit hacky but as Pierre said, I'm not sure if changing our fixtures would be the best option.

Reply:
Late to the party but I feel responsible for this 😛 

I agree with @ofek we should keep stubs as dumb as possible, the more logic we add, the higher the risk of hiding bugs and introduce tech debt. The problem here is that assertion logic shouldn't be here in the first place but I needed something quickly to make tests work again during the epic  Wheels Migration™️  and the Aggregator stub seemed like "if it fits, I sits".

So the quick-and-dirty solution if we don't want to touch fixtures would be moving special assertion logic elsewhere, better if we can keep it close to where it's actually needed like Ofek suggested. Even better (out of the scope of the PR) would be putting some work on the assertion logic, make it smarter (right now can be close to useless at times) and move it out of the stubs into a dedicated package with a clear API.